ModularTegustation/fishing/code/turfs.dm

```diff
@@ -63,6 +65,8 @@
 	. = ..()
 	if(!target_turf || is_type_in_typecache(thing, forbidden_types) || (thing.throwing && !istype(thing, /obj/item/food/fish || /obj/item/aquarium_prop )) || (thing.movement_type & (FLOATING|FLYING))) //replace this with a varient of chasm component sometime.
 		return
+	if(safe) // if this turf is in fact safe (currently only used for piscine mermaid)
+		return
```

Comment:
Move this above the type checks, but after the `. = ..()`. Putting it there reduces quantity of checks being made.

Reply:
i'll assume ya mean
```
/turf/open/water/deep/Entered(atom/movable/thing, atom/oldLoc) //Sinking Code
	. = ..()
	if(safe) // if this turf is in fact safe (currently only used for piscine mermaid)
		return
	if(!target_turf || is_type_in_typecache(thing, forbidden_types) || (thing.throwing && !istype(thing, /obj/item/food/fish || /obj/item/aquarium_prop )) || (thing.movement_type & (FLOATING|FLYING))) //replace this with a varient of chasm component sometime.
		return
```
if so, done